### DISCUSSION TITLE
0.45.0 beta

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#241] [Announcements]
- admin/home: update button
- added "get list" in folder menu #231
- folder menu clicking on last breadcrumb
- [plugins: own custom.html](https://github.com/rejetto/hfs/blob/main/dev-plugins.md#other-files)
- plugins: external URLs support for frontend_js and frontend_css
- new languages: French, Portuguese
- [admin/fs: "propagate" option when "see" permission is set on a folder](https://user-images.githubusercontent.com/1367199/236703891-29930d29-4d68-4c15-90d6-a3d9b853b1f9.png)
- [netmask with CIDR now supports operators](https://github.com/rejetto/hfs//wiki/Wildcards#network-masks)
- several fixes and minor improvements
- upload: "Skip existing files" option
- zip now includes empty folders
- session duration is now available in Admin/options page (env var method is now deprecated and will be removed in the future)
- several bugs fixed

beta12
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11496154/hfs-windows.zip)

beta11
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11490509/hfs-windows.zip)

beta10
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11486357/hfs-windows.zip)

alpha9
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11436616/hfs-windows.zip)

alpha7
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11436341/hfs-windows.zip)

alpha6
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11425851/hfs-windows.zip)

alpha4
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11421422/hfs-windows.zip)
[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/11421423/hfs-mac-arm.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/11421425/hfs-linux.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/11421426/hfs-mac.zip)

alpha1
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11416077/hfs-windows.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/11416078/hfs-linux.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/11416079/hfs-mac.zip)
[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/11416080/hfs-mac-arm.zip)

**Comment by mitobin** [comment]
Upload "Pick folder" doesn't work on mobile devices

**Reply by rejetto** [reply]
because it's not supported by mobiles

**Comment by rejetto** [comment]
those using npx can `npx -y hfs@beta`

**Reply by rejetto** [reply]
I just fixed a problem with npx on Windows. It was saving things to the wrong folder.

**Comment by rejetto** [comment]
alpha4
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11421422/hfs-windows.zip)
[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/11421423/hfs-mac-arm.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/11421425/hfs-linux.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/11421426/hfs-mac.zip)

**Comment by rejetto** [comment]
alpha6 with a couple of fixes
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11425851/hfs-windows.zip)

**Reply by xrivo1** [reply]
dovresti cercare di risolvere il problema dell'antivirus di windows. anche con questa versione da file infetto

**Reply by rejetto** [reply]
eh, ma non so bene come

**Reply by xrivo1** [reply]
capisco che non sia facile. comunque risolto anche stavolta permettendo l'esecuzione manualmente.

**Reply by xrivo1** [reply]
comunque giusto per tua informazione allego segnalazione di windows
![Trojan](https://github.com/rejetto/hfs/assets/106228981/89b54b8b-37c9-46df-ab64-6a30dddecd1a)

**Reply by rejetto** [reply]
sono quasi 10 anni che ricevo questo tipo di segnalazioni, purtroppo.
Anyway, running HFS using node as described below, shouldn't have any problem like that
https://github.com/rejetto/hfs#other-systems
I have plans for making a small exe that just runs HFS through Node.

**Comment by xrivo1** [comment]
La cosa strana é che fino allo 0.40.0. Non ha mai dato problemi. Il messaggio ha iniziato a comparire dalla 0.41.0

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by rejetto** [reply]
Quello che hai scritto mi ha fatto venire in mente una cosa.
Ho fatto qualche esperimento, e forse ho trovato come evitare il problema.
Mi puoi confermare che è risolto? [hfs-windows.zip](https://github.com/rejetto/hfs/files/11436341/hfs-windows.zip)

**Comment by rejetto** [comment]
alpha9
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11436616/hfs-windows.zip)

**Reply by xrivo1** [reply]
Perfetto. Nessuna segnalazione da parte di Windows defender. Installato con successo. :)

**Reply by rejetto** [reply]
![image](https://github.com/rejetto/hfs/assets/1367199/976d7ca1-8315-4aa7-9e28-799703ca6f18)

**Comment by mitobin** [comment]
is it possible to add online document viewing (office format)?

**Reply by rejetto** [reply]
this is the wrong place for requests.
Anyway, I'm not currently looking into that, so if anything comes it will probably be in a long time.
You should try browser extensions that do that.

**Reply by mitobin** [reply]
Thank you for guidance

**Comment by xrivo1** [comment]
Vedi? Bastava rimuovere il virus.
😂😂😂
Ovviamente scherzo !!
Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rejetto** [comment]
beta10
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11486357/hfs-windows.zip)

**Reply by GeorgeGeorgeGe** [reply]
Thanks for the new releases.
Windows Beta 10 crushes, i can not make it run.
Please check

**Reply by rejetto** [reply]
give me some info about it, like the error you get

**Reply by GeorgeGeorgeGe** [reply]
No errors, it opens and closes immediately in windows 10 and windows server 2019.
Thanks

**Reply by rejetto** [reply]
the error is in the windows that closes.
Run hfs from the command line, so you should be able to see the error.
Do you know how to do it?

**Reply by SanokKule** [reply]
Crashes when no config
```
PS C:\Other\hfs\test> .\hfs.exe
HFS ~ HTTP File Server - Copyright 2021-2023, Massimo Melina <a@rejetto.com>
License https://www.gnu.org/licenses/gpl-3.0.txt
started 5/16/2023, 17:20:51
version 0.45.0-beta10
build 2023-05-16T09:05:52.263Z
pid 30704
cwd C:\Other\hfs\test
config config.yaml
No config file, using defaults
pkg/prelude/bootstrap.js:1872
      throw error;
      ^

TypeError: (0 , exports.saveConfigAsap) is not a function
    at setConfig (C:\snapshot\hfs\dist\src\config.js:168:36)
    at failedOnFirstAttempt (C:\snapshot\hfs\dist\src\config.js:52:9)
    at install (C:\snapshot\hfs\dist\src\watchLoad.js:40:93)
    at watchLoad (C:\snapshot\hfs\dist\src\watchLoad.js:18:5)
    at Object.<anonymous> (C:\snapshot\hfs\dist\src\config.js:49:44)
    at Module._compile (pkg/prelude/bootstrap.js:1926:22)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
```

**Reply by GeorgeGeorgeGe** [reply]
no yaml included in the zip

**Reply by rejetto** [reply]
zip is not supposed to include yaml.
Thanks Sanokkule, I will work on that.

**Comment by rejetto** [comment]
sorry for the problem with beta10.
beta11
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11490509/hfs-windows.zip)

**Comment by xrivo1** [comment]
A me la beta 10 funziona benissimo.
Comunque grazie per la 11

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rejetto** [comment]
beta12 fixing 2 bugs
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11496154/hfs-windows.zip)

hopefully last beta before going official
